internal/daemon/api_plan.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"gopkg.in/yaml.v3"
+)
+
+func v1GetPlan(c *Command, r *http.Request, _ *userState) Response {
+	format := r.URL.Query().Get("format")
+	if format != "yaml" {
+		return statusBadRequest("invalid format %q", format)
+	}
+
+	servmgr := c.d.overlord.ServiceManager()
+	plan, err := servmgr.Plan()
+	if err != nil {
+		return statusInternalError("cannot get plan: %v", err)
```

Comment:
I think we can just forward the underlying error message here, since that operation ("getting the plan") is already what the method is supposed to do, so if we need to decorate an underlying naked error that should have happened inside it. It's of course not a big deal, but that kind of practice avoids repetitive error messages such as "cannot get plan: cannot generate plan: ..." or similar.

Reply:
Good call -- fixed.